api/src/test/users.test.ts

```diff
@@ -33,11 +38,27 @@ describe('Testing Users', () => {
           email: 'c@email.com',
           password: await bcrypt.hash('z1x2c3v4!', 10),
         },
-      ]);
+      ]
+
+      // Mock the findAllUser method
+      users.findAllUser = jest.fn().mockReturnValue(mockUsers);
+      (Sequelize as any).authenticate = jest.fn(); 
+
+      // Call the API for a get request
+      const result = await request(app.getServer()).get(`${usersRoute.path}`); 
+      
+      expect(result.status).toEqual(200);
+      expect(result.body.data.length).toEqual(3);
+      expect(result.body.data[0].id).toEqual(mockUsers[0].id); 
+      expect(result.body.data[0].email).toEqual(mockUsers[0].email); 
```

Comment:
I think using username is easier for now.

Reply:
Actually disregard. Lets use email, that way we can keep our user model the same.